src/site/_includes/partials/cards-collection.njk

```diff
@@ -1,10 +1,24 @@
 {% if cards %}
   {% for card in cards %}
     {% set featured = null %}
+    {% if card.isLatestPost %}
+      {% set latestPost = collections.blogPosts %}
+      {% if card.tags %}
+        {% set latestPost = collections.all | filterByTag(card.tags) %}
+      {% endif %}
+      {% set latestPostInTag = latestPost[0] %}
+      {% set featured = latestPostInTag.url | findByUrl %}
```

Comment:
I think I would move most of this logic to a filter named `latestPostByTags`, which uses `this.ctx.collections[tag]` to access the already created, memoized and sorted collections.